scrnaseq.Rmd

```diff
@@ -1674,142 +1748,400 @@ if (degs_found) {
+  degs_down = sapply(degs_down, function(n) seurat_rowname_to_entrez[[n]][1], USE.NAMES=TRUE, simplify=TRUE) %>% unlist() %>% as.character()
+  degs_down = degs_down[!is.na(degs_down)]
+  enrichr_results_down = EnrichrTest(genes=degs_down, databases=param$enrichr_dbs, padj=param$p_enrichr)
+
+  
+  # Flatten both enrichr results into tables
+  enrichr_results_up = purrr::map_dfr(names(enrichr_results_up), function(n) {
+    return(cbind(enrichr_results_up[[n]], 
+          list(Database=factor(rep(n, nrow(enrichr_results_up[[n]])), levels=names(enrichr_results_up)), 
+               Direction=factor(rep("up", nrow(enrichr_results_up[[n]])), levels=c("up", "down"))
+               )
+          ))
+  })
+  
+  enrichr_results_down = purrr::map_dfr(names(enrichr_results_down), function(n) {
+    return(cbind(enrichr_results_down[[n]], 
+          list(Database=factor(rep(n, nrow(enrichr_results_down[[n]])), levels=names(enrichr_results_down)), 
+               Direction=factor(rep("up", nrow(enrichr_results_down[[n]])), levels=c("up", "down"))
+               )
+          ))
+  })
+  
+  # Rbind and add factor levels
+  enrichr_results = dplyr::bind_rows(enrichr_results_up, enrichr_results_down)
+  return(c(contrast, list(enrichr=enrichr_results)))
+  
+  # Wait two second for server
+  Sys.sleep(2)
+})
+
+# Now regroup list so that subsets are together again
+original_contrast_rows = purrr::map_int(degs_contrasts_results, function(contrast){return(contrast[["contrast_row"]]) })
+degs = split(degs_contrasts_results, original_contrast_rows)
+
+# Write degs to files
+degs_result_files = purrr::map_chr(degs, function(degs_subsets) {
+  # output file
+  file = paste0(param$path_out, "/degs_contrast_row_", degs_subsets[[1]][["contrast_row"]], "_results.xlsx")
+  
+  # degs
+  degs_subsets_results = purrr::map(degs_subsets, function(contrast) {return(contrast[["results"]])})
+  names(degs_subsets_results) = purrr::map_chr(degs_subsets, function(contrast) {return(ifelse(!is.na(contrast[["subset_group"]]), contrast[["subset_group"]], "All"))})
+  file = DegsWriteToFile(degs_subsets_results, annot_ensembl=annot_ensembl, file=file)
+  
+  return(file)
+})
+
+# Write enrichr results to files
+degs_enrichr_files = purrr::map_chr(degs, function(degs_subsets) {
+  # output file
+  file = paste0(param$path_out, "/degs_contrast_row_", degs_subsets[[1]][["contrast_row"]], "_functions.xlsx")
+  
+  # enrichr results
+  degs_subsets_enrichr = purrr::map(degs_subsets, function(contrast) {return(contrast[["enrichr"]])})
+  names(degs_subsets_enrichr) = purrr::map_chr(degs_subsets, function(contrast) {return(ifelse(!is.na(contrast[["subset_group"]]), contrast[["subset_group"]], "All"))})
+  file = EnrichrWriteResults(degs_subsets_enrichr, file=file)
+  
+  return(file)
+})
+
+```
+
+```{r deg_summary, results="asis", fig.show="asis"}
+knitr_header_string = '
+
+### {{condition_column}}: {{condition_group1}} vs {{condition_group2}}
```

Comment:
I guess this one I have to visually see to imagine how it looks like.

Reply:
Yes it is a template markdown which then will be filled out with values of variables.